tests/s3/versioning/test_versioning_delete_objects.py

```diff
@@ -0,0 +1,483 @@
+from libs.s3.s3_test_lib import S3TestLib
+from libs.s3.s3_versioning_test_lib import S3VersioningTestLib
+from libs.s3.s3_multipart_test_lib import S3MultipartTestLib
+from libs.s3.s3_versioning_common_test_lib import check_get_head_object_version
+from libs.s3.s3_versioning_common_test_lib import check_list_object_versions
+from libs.s3.s3_versioning_common_test_lib import check_list_objects
+from libs.s3.s3_versioning_common_test_lib import delete_objects
+from libs.s3.s3_versioning_common_test_lib import upload_version
+from libs.s3.s3_versioning_common_test_lib import upload_versions
+
+
+class TestVersioningDeleteObjects:
+    """Test DELETE Objects API with Object Versioning"""
+
+    # pylint:disable=attribute-defined-outside-init
+    # pylint:disable-msg=too-many-instance-attributes
+    def setup_method(self):
+        """Function will be perform prerequisite test steps prior to each test case."""
+        self.log = logging.getLogger(__name__)
+        self.log.info("STARTED: Setup operations")
+        self.s3_test_obj = S3TestLib(endpoint_url=S3_CFG["s3_url"])
+        self.s3_ver_test_obj = S3VersioningTestLib(endpoint_url=S3_CFG["s3_url"])
+        self.s3mp_test_obj = S3MultipartTestLib()
+        self.test_dir_path = os.path.join(TEST_DATA_FOLDER, "TestVersioningDeleteObject")
+        if not path_exists(self.test_dir_path):
+            make_dirs(self.test_dir_path)
+            self.log.info("Created path: %s", self.test_dir_path)
+        file_name1 = "{0}{1}".format("ver_del_obj", time.perf_counter_ns())
+        file_name2 = "{0}{1}".format("ver_del_obj", time.perf_counter_ns())
+        file_name3 = "{0}{1}".format("ver_del_obj", time.perf_counter_ns())
+        mp_file_name = "{0}{1}".format("ver_del_obj_mpu", time.perf_counter_ns())
+        download_file = "{0}{1}".format("ver_download", time.perf_counter_ns())
+        self.file_path1 = os.path.join(self.test_dir_path, file_name1)
+        self.file_path2 = os.path.join(self.test_dir_path, file_name2)
+        self.file_path3 = os.path.join(self.test_dir_path, file_name3)
+        self.mp_file_path = os.path.join(self.test_dir_path, mp_file_name)
+        self.download_path = os.path.join(self.test_dir_path, download_file)
+        self.upload_file_paths = [self.file_path1, self.file_path2, self.file_path3]
+        for file_path in self.upload_file_paths:
+            create_file(file_path, 1, "/dev/urandom")
+            self.log.info("Created file: %s", file_path)
+        self.bucket_name = "ver-bkt-{}".format(time.perf_counter_ns())
+        self.object_name1 = "obj-1-{}".format(time.perf_counter_ns())
+        self.object_name2 = "obj-2-{}".format(time.perf_counter_ns())
+        self.object_name3 = "obj-3-{}".format(time.perf_counter_ns())
+        self.object_name4 = "obj-4-{}".format(time.perf_counter_ns())
+        self.object_name5 = "obj-5-{}".format(time.perf_counter_ns())
+        self.object_name6 = "obj-6-{}".format(time.perf_counter_ns())
```

Comment:
can we rather loop around for creating objects depending on number of objects to be created maybe defined in yaml file?

Reply:
Was contemplating making this a list of objects, but then I felt it would reduce readability of the tests, so stuck with this.